### PR TITLE
(WIP) Add proxy server into oauth2 product test env

### PR DESCRIPTION
**Pull request by sergey-melnychuk**

### CODE REVIEW

File: testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2/config.properties [hunk 1]
```diff
@@ -23,3 +23,5 @@ http-server.authentication.oauth2.client-id=trinodb_client_id
 http-server.authentication.oauth2.client-secret=trinodb_client_secret
 http-server.authentication.oauth2.user-mapping.pattern=(.*)(@.*)?
 oauth2-jwk.http-client.trust-store-path=/docker/presto-product-tests/conf/presto/etc/hydra/cert/localhost.pem
+#http-client.http-proxy=proxy:12345
```
Comment (sergey-melnychuk) [inline testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2/config.properties:26]: Not sure how to propagate the `http-client.*` properties required by airlift's `HttpClientConfig`.

If I uncomment them here I get error:

```
presto-master       | Errors:
presto-master       | 
presto-master       | 1) Configuration property 'http-client.http-proxy' was not used
presto-master       | 2) Configuration property 'http-client.http-proxy.secure' was not used
```

Comment (lukasz-walkiewicz) [reply testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2/config.properties:26]: Try:
```properties
oauth2-jwk.http-client.http-proxy=proxy:12345
```

Comment (lukasz-walkiewicz) [reply testing/trino-product-tests-launcher/src/main/resources/docker/presto-product-tests/conf/environment/singlenode-oauth2/config.properties:26]: `oauth2-jwk` is the name and also the prefix of the http client

**Comment by sergey-melnychuk** [comment]
https://github.com/trinodb/trino/pull/10069

**Pull request closed without merging**